### ISSUE TITLE
Unable to Compile with ofx4j on Java 17

### ISSUE THREAD

**Issue by Jaco-Pretorius** [stoicflame/ofx4j#71]
When trying to compile with the ofx4j library (version 1.35) I get the following error

```
error reading /Users/{myuser}/.gradle/caches/modules-2/files-2.1/com.webcohesion.ofx4j/ofx4j/1.35/d3d08d31b36d698f3f2f02174274359c618efd62/ofx4j-1.35.jar; Invalid CEN header (invalid zip64 extra data field size)
```

This seems like the exact same issue filed on [ApkTool](https://github.com/iBotPeaches/Apktool/issues/3174) 
From that thread it seems like the issue is caused by JDK 17.0.8.

The following is mentioned [in the JDK changelog](https://www.oracle.com/java/technologies/javase/17all-relnotes.html#JDK-8302483):

> Improved ZIP64 Extra Field Validation (JDK-8302483 (not public))
> 
> java.util.zip.ZipFile has been updated to provide additional validation of ZIP64 extra fields when opening a ZIP file. This validation may be disabled by setting the system property jdk.util.zip.disableZip64ExtraFieldValidation to true.

I am trying to disable that validation locally (I haven't figure out the correct Gradle config for that yet), but it seems like the root of the issue is that the ofx4j jar itself is failing a ZIP file validation.

**Comment by stoicflame** [comment]
Thanks for the report; I'll take a look.

**Comment by jpstotz** [comment]
I know that there are a lot of Android APK/ZIP files which run into the "Invalid CEN header" problem, but this is the first time I see a JAR file causing it.  I am in contact with the developer Lance Andersen from Oracle who wrote the JDK-8302483. He is especially interested in the tool chain which creates ZIP/JAR files that cause the "Invalid CEN header" error in the new JDK versions. 

A self-built version of your project (using an older JDK 8 version) creates an ofx4j-1.36-SNAPSHOT.jar file that causes the "Invalid CEN header". As your project is not very large and easy to build I will point Lance to your project as an affected example project  that suffers from massive damage by JDK-8302483.

**Comment by sea-steve** [comment]
I started seeing the same problem today in a maven project with a dependency on ofx4j. It completely blocked my build for most of the day. It took quite some time to understand the problem, since most Google results discussing the "Invalid CEN header" error simply recommend deleting the (corrupt) local jar file and re-downloading it, but of course that didn't fix the problem in this case, since the jar isn't actually corrupt. I wasn't able to find a way to turn off this check in Maven. (Has anyone else?) But, since apparently the problematic JDK change was made in JDK 17 and only backported to JDK 11 (so far), I was able to temporarily workaround the issue by installing and compiling my project with JDK 19, which doesn't exhibit the issue. But I imagine not everyone will be able to easily do this...

**Comment by stoicflame** [comment]
Can anyone suggest the best way to resolve this? 

I'm happy to deploy using a different version of JDK 8, but would that fix the problem? How can I best reproduce this to ensure a fix?

The other option is to just drop support for Java 8 and require Java 11? I'm not sure how many developers that would effect.

**Comment by jpstotz** [comment]
You have to set the property `jdk.util.zip.disableZip64ExtraFieldValidation=true`, so for a maven project the following command-line should work e.g. for executing `package`:

    mvn package -DargLine="-Djdk.util.zip.disableZip64ExtraFieldValidation=true"

**Comment by stoicflame** [comment]
@jpstotz do you have a suggestion for what to do in the ofx4j project to create a jar that doesn't manifest this problem?

**Comment by jpstotz** [comment]
@stoicflame As only a few projects are affected by this problems one can assume that JAR files created by Maven without any plugins don't have the "Invalid CEN header" problem. therefore one of the used plugins my be the cause.

Looking at the pom.xml ofx4j there aren't many possible plugins that modify the JAR file. I assume that `nexus-staging-maven-plugin` and `maven-javadoc-plugin` do not modify the JAR file.

So left over are `maven-gpg-plugin` and `maven-bundle-plugin`. I would disable the pgp plugin for tests and check the created JAR file (just try to open it in a `java.util.zip.ZipFile` using the latest java 17 version and you will see if the JAR is affected or not. 

In a quick test I upgraded to `maven-bundle-plugin` to 5.1.9 and the created JAR file did not had any problems. It seems like all versions greater or equal to 5.1.5 create JAR files that are not affected by the "Invalid CEN header" problem.

I have created an historical issue in their issue tracker to make it easier for other projects to fix a proper fix: https://issues.apache.org/jira/browse/FELIX-6622
